src/std/iter.ss

```diff
@@ -139,23 +139,26 @@ package: std
   ((count start) (iter-in-iota start count 1))
   ((count start step) (iter-in-iota start count step)))
 
-(def (iter-in-range start end step)
+(def (iter-in-range start end (step 1) (cmp <))
   (def (next it)
     (with ((iterator e) it)
-      (if (< e end)
+      (if (cmp e end)
         (begin
           (set! (&iterator-e it) (+ e step))
           e)
         iter-end)))
-  (if (and (real? start) (real? end) (real? step))
-    (make-iterator start next)
-    (error "Parameters are of wrong type (start:real end:real step:real)."
-      [start end step])))
+  (make-iterator start next))
 
 (def* in-range
-  ((end) (iter-in-range 0 end 1))
-  ((start end) (iter-in-range start end 1))
-  ((start end step) (iter-in-range start end step)))
+  ((end) (iter-in-range 0 end))
+  ((start end)
+   (if (> start end)
+     (iter-in-range start end -1 >)
+     (iter-in-range start end)))
```

Comment:
Here `step` is 1 and `cmp` is <, but that's implicit. I think explicit is better in this case.